### PR TITLE
Migrate cluster-api-addon-provider-helm presubmits to community infra

### PR DESCRIPTION
**Pull request by varaynya**
migrate  pull-cluster-api-addon-provider-helm-e2e, -e2e-workload-upgrade, -apiversion-upgrade to community cluster #31791

### CODE REVIEW

File: config/jobs/kubernetes-sigs/cluster-api-addon-provider-helm/cluster-api-addon-provider-helm-presubmits-main.yaml [hunk 1]
```diff
@@ -156,6 +157,13 @@ presubmits:
             value: \[REQUIRED\]
           - name: GINKGO_SKIP
             value: ""
+        resources:
+          limits:
+            cpu: 2
+            memory: 4Gi
+          requests:
+            cpu: 2
+            memory: 4Gi
```
Comment (ameukam) [inline config/jobs/kubernetes-sigs/cluster-api-addon-provider-helm/cluster-api-addon-provider-helm-presubmits-main.yaml:166]:
```suggestion
            memory: 4Gi
```

File: config/jobs/kubernetes-sigs/cluster-api-addon-provider-helm/cluster-api-addon-provider-helm-presubmits-main.yaml [hunk 2]
```diff
@@ -226,6 +235,13 @@ presubmits:
           - name: GINKGO_FOCUS
             value: "\\[K8s-Upgrade\\]"
         # we need privileged mode in order to do docker in docker
+        resources:
+          limits:
+            cpu: 2
+            memory: 4Gi
+          requests:
+            cpu: 2
+            memory: 4Gi
```
Comment (ameukam) [inline config/jobs/kubernetes-sigs/cluster-api-addon-provider-helm/cluster-api-addon-provider-helm-presubmits-main.yaml:244]:
```suggestion
            memory: 4Gi
```

File: config/jobs/kubernetes-sigs/cluster-api-addon-provider-helm/cluster-api-addon-provider-helm-presubmits-main.yaml [hunk 3]
```diff
@@ -260,6 +277,13 @@ presubmits:
             - name: GINKGO_FOCUS
               value: "API Version Upgrade"
           # we need privileged mode in order to do docker in docker
+          resources:
+            limits:
+              cpu: 2
+              memory: 4Gi
+            requests:
+              cpu: 2
+              memory: 4Gi
```
Comment (ameukam) [inline config/jobs/kubernetes-sigs/cluster-api-addon-provider-helm/cluster-api-addon-provider-helm-presubmits-main.yaml:286]: `resources` is already present on L265. Update the existing resources.

**Comment by ameukam** [comment]
/retitle Migrate cluster-api-addon-provider-helm presubmits to community infra

**Comment by varaynya** [comment]
Hi Arnaud,
Should I just remove the resources section for
"pull-cluster-api-addon-provider-helm-e2e,
-e2e-workload-upgrade, -apiversion-upgrade" jobs?

**Comment by ameukam** [comment]
@varaynya it needs to be updated.
See: https://github.com/kubernetes/test-infra/blob/f2907b59990f5c11628134b1b792d10df46ac7fc/config/jobs/kubernetes-sigs/cluster-api-addon-provider-helm/cluster-api-addon-provider-helm-presubmits-main.yaml#L265-L267

File: config/jobs/kubernetes-sigs/cluster-api-addon-provider-helm/cluster-api-addon-provider-helm-presubmits-main.yaml [hunk 4]
```diff
@@ -225,16 +234,21 @@ presubmits:
         env:
           - name: GINKGO_FOCUS
             value: "\\[K8s-Upgrade\\]"
-        # we need privileged mode in order to do docker in docker
+        # we need privileged mode in order to do docker in docker
```
Comment (ameukam) [inline config/jobs/kubernetes-sigs/cluster-api-addon-provider-helm/cluster-api-addon-provider-helm-presubmits-main.yaml:237]:
```suggestion
        # we need privileged mode in order to do docker in docker
```

**Pull request merged**